tensorflow_lite_support/python/task/text/nl_classifier.py

```diff
@@ -0,0 +1,142 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NL Classifier task."""
+
+import dataclasses
+from typing import Optional
+
+from tensorflow_lite_support.python.task.core import task_options
+from tensorflow_lite_support.python.task.core import task_utils
+from tensorflow_lite_support.python.task.processor import processor_options
+from tensorflow_lite_support.python.task.text.pybinds import _pywrap_nl_classifier
+from tensorflow_lite_support.python.task.text.pybinds import nl_classifier_options_pb2
+
+_ProtoNLClassifierOptions = nl_classifier_options_pb2.NLClassifierOptions
+_CppNLClassifier = _pywrap_nl_classifier.NLClassifier
+_NLClassificationOptions = processor_options.NLClassificationOptions
+_Category = processor_options.Category
+_Classifications = processor_options.Classifications
+_ClassificationResult = processor_options.ClassificationResult
+
+
+@dataclasses.dataclass
+class NLClassifierOptions:
+  """Options for the NL classifier task."""
+  base_options: task_options.BaseOptions
+  nl_classification_options: Optional[_NLClassificationOptions] = None
+
+
+class NLClassifier(object):
+  """Class that performs NL classification on text."""
+
+  def __init__(self, options: NLClassifierOptions,
+               cpp_classifier: _CppNLClassifier) -> None:
+    """Initializes the `NLClassifier` object."""
+    # Creates the object of C++ NLClassifier class.
+    self._options = options
+    self._classifier = cpp_classifier
+
+  @classmethod
+  def create_from_file(cls, file_path: str) -> "NLClassifier":
+    """Creates the `NLClassifier` object from a TensorFlow Lite model.
+
+    Args:
+      file_path: Path to the model.
+
+    Returns:
+      `NLClassifier` object that's created from the model file.
+    Raises:
+      ValueError: If failed to create `NLClassifier` object from the provided
+        file such as invalid file.
+      RuntimeError: If other types of error occurred.
+    """
+    model_file = task_options.ExternalFile(file_name=file_path)
+    base_options = task_options.BaseOptions(model_file=model_file)
+    options = NLClassifierOptions(base_options=base_options)
+    return cls.create_from_options(options)
+
+  @classmethod
+  def create_from_options(cls, options: NLClassifierOptions) -> "NLClassifier":
+    """Creates the `NLClassifier` object from NL classifier options.
+
+    Args:
+      options: Options for the NL classifier task.
+
+    Returns:
+      `NLClassifier` object that's created from `options`.
+    Raises:
+      ValueError: If failed to create `NLClassifier` object from
+        `NLClassifierOptions` such as missing the model.
+      RuntimeError: If other types of error occurred.
+    """
+    proto_options = _ProtoNLClassifierOptions()
+    proto_options.base_options.CopyFrom(
+      task_utils.ConvertToProtoBaseOptions(options.base_options))
+
+    if options.nl_classification_options:
```

Comment:
@khanhlvg I'm not sure how to use the Python dataclass in the pybind layer, so I've followed one of the [older](https://github.com/tensorflow/tflite-support/blob/c8d6415e2fec084b18179428c71c3be2a53ae556/tensorflow_lite_support/python/task/vision/image_classifier.py#L109) patterns that explicity sets the classification options in the `create_from_options` classmethod.